osf/models/metaschema.py

```diff
@@ -79,3 +79,8 @@ def validate_metadata(self, metadata, reviewer=False, required_fields=False):
         except jsonschema.SchemaError as e:
             raise ValidationValueError(e.message)
         return
+
+
+class FileMetadataSchema(AbstractSchema):
+
+    serializer_class = None
```

Comment:
Hmm, thinking about this again, I'm not sure this idea will work as we were hoping. 

In order for each metadata schema instance to have it's own `serializer_class`, we would need to make `FileMetadataSchema` an abstract class and create a table for each schema type.
```
class FileMetadataSchema(AbstractSchema):
	
	serializer_class = None

	class Meta:
		abstract = True

class DataciteFileMetadataSchema(FileMetadataSchema):
	
	serializer_class = DataciteRecordSerializer

class CrossrefMetadataSchema(FileMetadataSchema):
	
	serializer_class = CrossrefRecordSerializer
```

Alternatively, we could make the `serializer_class` a real field on the `FileMetadataSchema` model (not actually sure if this would work, just spitballing here).
```
class FileMetadataSchema(AbstractSchema):
	
	serializer_class = models.TextField(...)
```

Alternatively, we could use the approach roughly implemented in #8354 ([here](https://github.com/caseyrollins/osf.io/blob/ed433bd2a5dd4673619662fe3e38926b763e2128/osf/models/metaschema.py#L33-L37))
```
class FileMetadataMetaSchema(AbstractMetaSchema):

    @property
    def formatter(self):
        for formatter in FileMetadataFormatter.__subclasses__():
            if formatter.metaschema_id == self._id:
                return formatter
```

I'm genuinely not sure which approach would be best or if there are alternatives we're missing. Might need to rope @sloria into a real life chat about this.

Reply:
I am down for real life chatting! Seems like it'll be useful!

Reply:
Initial questions:

1. What are the serializers used for? Are they what do the work of combining auto-generated and user-generated metadata?
2. Where will the serializers be defined?
3. When will the serializers be used? Would we have to make an additional query for a FileMetadataRecord's schema just to get the serializer?

```python
file_metadata_record.schema.serializer_class  # unnecessary join?
```

Alternative #3: Use a registry, like we do for [file event notifications](https://github.com/CenterForOpenScience/osf.io/blob/092c996695fdb2b43c5307a751d764d6482b73a6/website/notifications/events/base.py)

```python
@register(metaschema_id='datacite')
class DataciteMetadataSerializer(Serializer):
    ...


registry['datacite']  # DataciteMetadataSerializer
```